packages/client/src/stores/notification.js

```diff
@@ -37,26 +39,32 @@ const createNotificationStore = () => {
       })
       return
     }
-
-    store.set({
-      id: generate(),
-      type,
-      message,
-      icon,
-      dismissable: !autoDismiss,
-      delay: get(store) != null,
+    const _id = id()
+    store.update(state => {
+      return [
+        ...state,
+        {
+          id: _id,
+          type,
+          message,
+          icon,
+          dismissable: !autoDismiss,
+          delay: get(store) != null,
+        },
+      ]
     })
-    clearTimeout(timeout)
     if (autoDismiss) {
-      timeout = setTimeout(() => {
-        store.set(null)
+      const timeoutId = setTimeout(() => {
```

Comment:
As above, we don't need to worry about storing a reference. We can just call `setTimeout`.